sql/core/src/main/scala/org/apache/spark/sql/GroupedData.scala

```diff
@@ -34,15 +65,32 @@ import org.apache.spark.sql.types.NumericType
  * @since 1.3.0
  */
 @Experimental
-class GroupedData protected[sql](df: DataFrame, groupingExprs: Seq[Expression]) {
-
+class GroupedData protected[sql](
+    df: DataFrame,
+    groupingExprs: Seq[Expression],
+    private val groupType: GroupedData.GroupType) {
   private[sql] implicit def toDF(aggExprs: Seq[NamedExpression]): DataFrame = {
```

Comment:
Actually used multiple places, e.g. in `agg()`, `count()` and `aggregateNumericColumns()`